docs/extending-nsm.md

```diff
@@ -0,0 +1,129 @@
+# Extending Network Service Mesh (NSM)
```

Comment:
I wonder if this shouldn't be named Network Service Mesh Developer's Guide or Network Service Mesh Extensions Developer's Guide.

Reply:
I think I should withdraw this comment but I am still wondering about the scope of this document the example described is a dataplane extension. Will this procedure also apply to any other plugins?